doc/0a-primer.md

```diff
@@ -0,0 +1,134 @@
+These tools include Kraken and mifaser. 
+5. **Use assembly graphs ([compact] de Bruijn graphs) to rescue unassembled and unbinned reads that are not in reference databases, and analyze these reads**
+Assembly graphs contain all of the sequences that are contained in the reads. 
+Using assembly graphs, it is possible to access and analyze unassembled and unbinned reads that have not been previously deposited in a reference database.
+Many tools facilitate assembly graph analysis, including the assembly graph builder [BCALM](https://github.com/GATB/bcalm) and the assembly graph visualizer [Bandage](https://rrwick.github.io/Bandage/).
+Spacegraphcats introduces a computationally efficient framework for organizing and querying assembly graphs at scale.
+
+Below we introduce the concept of assembly graphs and their utility in metagenome analysis.
+
+## Representing all the k-mers: de Bruijn graphs and compact de Bruijn graphs
+
+In real life, DNA sequences are fully contiguous within a genome (or chromosome).
+When a genome is chopped up into little pieces for sequencing, connectivity information is lost.
+In order to get this connectivity information back to create an assembly, assemblers use different strategies to identify overlapping sequences among reads.
+
+![](https://i.imgur.com/1ZS1WNe.png) *Figure from Ayling et al. 2020 (https://doi.org/10.1093/bib/bbz020) depicting two strategies commonly used by assemblers.*
+
+A de Bruijn graph is one way to represent this information.
+A de Bruijn graph is built by breaking sequencing data down into k-mers and finding all overlaps of size k-1 between all k-mers.
+The figure above below demonstrates this process.
+Each node in the graph represents a k-mer (here of size 3), and each arrow represents an overlap between two k-mer by *k*-1 nucleotides (here, by 2 nucleotides).
+Each k-mer only occurs in the graph once.
+
+A compact de Bruijn graph (cDBG) is built from the de Bruijn graph by collapsing all linear paths of k-mers into a single node.
+While all the nodes in this graph happen to be the same length, nodes in a cDBG usually vary in size.
+Every k-mer still only occurs once in the graph, but now nodes grow in size to be greater than *k*.
+Annecdotally, nodes in a metagenome cDBG range from *k* to ~2000 nucleotides when *k* = 31.
+
+![](https://i.imgur.com/a74QkYI.jpg) *de Bruijn graph collapsed to a compact de Bruijn graph.*
+
+Both DBGs and cDBGs contain every k-mer from a sample.
+
+## Assembly graphs in the wild
+
+Real metagenome assembly graphs are complex.
+Below is an assembly graph from a mouse gut metagenome by @SilasKieser. 
+Many metagenome graphs are fully connected (e.g. have a single component), and are large and complex.
+
+![](https://i.imgur.com/TEJ3zJZ.png) *An assembly graph of a mouse gut metagenome. The assembly graph was built by metaspades and visualized with Bandage. Image courtesy of Silas Kieser.*
```

Comment:
do we want to leave these as imgur URLs? they may eventually break. maybe better to have them in the repo?

Reply:
ya i didn't know what to do here...it was easier to just use the png urls e.g. that are created when a png is uploaded to hackmd, but I can switch them over if you prefer.

Reply:
is ok for now, just make an issue when we merge this, please!